webclient/app/src/app/commons/templateDialog/TemplateDialog.jsx

```diff
@@ -0,0 +1,65 @@
+import React, {useEffect} from "react";
+import {
+    Button,
+    Box,
+    Dialog,
+    DialogActions,
+    DialogTitle,
+    IconButton,
+    TextField,
+    Typography
+} from "@mui/material";
+import LoadingSpinner from "../loadingSpinner/LoadingSpinner";
+import TemplateDialogStyles from "./TemplateDialogStyles";
+import {useTranslation} from "react-i18next";
+import {Close} from "@mui/icons-material";
+
+function TemplateDialog(props) {
+
+    let [appTemplate, setAppTemplate] = React.useState(null);
+    const templateDialogStyles = TemplateDialogStyles();
+    const {t} = useTranslation();
+
+    const {appTemplateId, onClose, handleSave, appTemplates} = props;
+
+    useEffect(() => {
+        setAppTemplate({...appTemplates.find(appTemplate => appTemplate.id === appTemplateId)})
+    }, [appTemplateId, appTemplates])
+    
+    if (!appTemplate) {
```

Comment:
It would be great, if you could differentiate between all appTemplates are loading and no appTemplate is selected. I think you are always displaying a loading bar, even if the appTemplates are loaded.